internal/ls/inlay_hints.go

```diff
@@ -27,6 +27,9 @@ func (l *LanguageService) ProvideInlayHint(
 ) (lsproto.InlayHintResponse, error) {
 	program, file := l.getProgramAndFile(params.TextDocument.Uri)
 	quotePreference := getQuotePreference(file, l.UserPreferences())
+	if !isAnyInlayHintEnabled(l.UserPreferences()) {
+		return lsproto.InlayHintsOrNull{InlayHints: nil}, nil
+	}
```

Comment:
This could probably get moved a couple of lines up.

(setting the field to nil is also redundant, but that's ok)